utils/extractor/src/index.ts

```diff
@@ -293,3 +293,53 @@ export type SheetCapture = {
   data: Flatfile.RecordData[]
   metadata?: { rowHeaders: number[] }
 }
+
+export function parseSheet(jsonArray: any[]): SheetCapture {
+  try {
+    // Ensure all items are objects
+    const filteredResults = jsonArray.filter(
+      (item) => typeof item === 'object' && item !== null
+    )
+
+    if (filteredResults.length === 0) {
+      return {} as SheetCapture
+    }
+
+    // Custom flatten function
+    const flattenObject = (obj: any, parent: string = '', res: any = {}) => {
```

Comment:
Personal preference is not mutating argument inputs but creating a copy of the input  and returning the mutated version.